test/unit/cute/intel_xe/copy_1d.cpp

```diff
@@ -0,0 +1,296 @@
+
+  auto thr_copy_ldsm = tiled_ldsm.get_thread_slice(syclcompat::local_id::x());
+  auto thr_copy_stsm = tiled_stsm.get_thread_slice(syclcompat::local_id::x());
+
+  Tensor thr_tile_load_S =
+      thr_copy_load.partition_S(tile_S); // (CopyOp, CopyM, CopyN)
+  Tensor thr_tile_store_D =
+      thr_copy_store.partition_D(tile_D); // (CopyOp, CopyM, CopyN)
+
+  Tensor thr_tile_ldsm_S =
+      thr_copy_ldsm.partition_S(sTensor); // (CopyOp, CopyM, CopyN)
+  Tensor thr_tile_stsm_D =
+      thr_copy_stsm.partition_D(sTensor); // (CopyOp, CopyM, CopyN)
+
+  // Construct a register-backed Tensor with the same shape as each thread's
+  // partition Use make_fragment because the first mode is the instruction-local
+  // mode
+  Tensor fragment = make_fragment_like(
+      thr_copy_load.partition_D(tile_S)); // (CopyOp, CopyM, CopyN)
+
+#if 0
+  if (thread(0, 0)) {
+    print("loading to registers from src ========================\n");
+    print("tile_S:");
+    print(tile_S.layout());
+    print("\n");
+
+    print("thr_tile_load_S: ");
+    print(thr_tile_load_S.layout());
+    print("\n");
+    print(thr_tile_load_S.data());
+    print("\n");
+
+    print("thr_tile_store_D: ");
+    print(thr_tile_store_D.layout());
+    print("\n");
+
+    print("fragment: ");
+    print(fragment.layout());
+    print("\n");
+  }
+#endif
+
+  // Copy from GMEM to RMEM and from RMEM to GMEM
+  prefetch(tiled_copy_load, thr_tile_load_S);
+  copy(tiled_copy_load, thr_tile_load_S, fragment);
+  copy(tiled_stsm, fragment, thr_tile_stsm_D);
+  clear(fragment);
+  copy(tiled_ldsm, thr_tile_ldsm_S, fragment);
+  copy(tiled_copy_store, fragment, thr_tile_store_D);
+}
+
+TEST(PVC_1d_copy, copy_double) {
+  {
+    constexpr int M = 1;
+    constexpr int N = 128;
+    using Element = double;
+    //
+    // Allocate and initialize
+    //
+    std::vector<Element> host_src(M * N);
+    std::vector<Element> host_output(M * N);
+
+    Element *device_src = syclcompat::malloc<Element>(M * N);
+    Element *device_output = syclcompat::malloc<Element>(M * N);
+
+    for (size_t i = 0; i < host_src.size(); ++i) {
+      host_src[i] = static_cast<Element>(i);
+    }
+
+    syclcompat::memcpy<Element>(device_src, host_src.data(), M * N);
+    syclcompat::memcpy<Element>(device_output, host_output.data(), M * N);
+
+    Tensor S =
+        make_tensor(make_gmem_ptr(device_src),
+                    make_layout(Shape<Int<M>, Int<N>>{}, Stride<Int<N>, _1>{}));
+    Tensor D =
+        make_tensor(make_gmem_ptr(device_output),
+                    make_layout(Shape<Int<M>, Int<N>>{}, Stride<Int<N>, _1>{}));
+
+    static constexpr auto subgroup_size = 16;
+    auto blockDim = syclcompat::dim3(subgroup_size);
+
+    launch<copy_kernel_vectorized<decltype(S), decltype(D)>>(
+        launch_policy{
+            syclcompat::dim3(1), blockDim,
+            local_mem_size{static_cast<std::size_t>(0)},
+            kernel_properties{sycl_exp::sub_group_size<SUBGROUP_SIZE>}},
```

Comment:
You do not need to pass the `local_mem_size` if you do not need local memory from SYCLCompat, as in, you use the intel local memory extension to allocate local memory, hence you do not need to pass `local_mem_size` here. As fas as syclcompat is concerned,  you need not tell it that you will be using local memory later